docs/architecture/grpc-for-wcf-developers/authentication.md

```diff
@@ -0,0 +1,18 @@
+---
+title: Authentication and Authorization
+description: gRPC for WCF Developers | Authentication and Authorization
+author: markrendle
+ms.date: 09/02/2019
+---
+
+# Authentication and authorization
+
+gRPC authentication works on two levels. Channel-level authentication is based on SSL/TLS certificates and is applied when the channel is opened. Call-level authentication is handled using tokens which are applied in metadata when the call is made. You can use either or both of these mechanisms to secure your service.
+
+> [!NOTE]
+> Because gRPC uses HTTP/2, which requires SSL/TLS, all traffic between clients and servers is already encrypted, even if you don't use channel-level authentication.
```

Comment:
HTTP/2 doesn't require SSL/TLS. Browsers require it, but you can send unencrypted HTTP/2 between many non-browser HTTP/2 clients and servers.

e.g. https://docs.microsoft.com/en-us/aspnet/core/grpc/troubleshoot?view=aspnetcore-3.0#unable-to-start-aspnet-core-grpc-app-on-macos

Every gRPC client and server implementation supports HTTP/2 without encryption.

Reply:
TIL